src/zenml/pipelines/base_pipeline.py

```diff
@@ -147,6 +147,7 @@ def _verify_arguments(self, *steps: BaseStep, **kw_steps: BaseStep) -> None:
             )
 
         combined_steps = {}
+        step_refs: Dict[type, str] = {}
```

Comment:
I think a set is enough in this case as we don't actually use the dict values as far as I can see. I'd also try to be a little be more explicit with the actual type annotation of the set, in our case it can only contain `BaseStep` subclasses: 
```python
step_classes = Set[Type[BaseStep]] = set()
```

Reply:
okay, even I had second thoughts about this, I will change it in the next commit.